### PR TITLE
[DD4hep] Children by Name and CopyNo on a Path

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* This PR is a test to check if this is what @vargasa needs for MaterialBudget

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by cvuosalo** [comment]
Why are there comparison differences?

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 1]
```diff
@@ -194,13 +213,104 @@ bool DDFilteredView::firstChild() {
...
   }
   LogVerbatim("DDFilteredView") << "Search for first child failed.";
   return false;
 }
 
+int DDFilteredView::nodeCopyNo(const std::string_view copyNo) const {
+  int result;
+  if (auto [p, ec] = std::from_chars(copyNo.data(), copyNo.data() + copyNo.size(), result); ec == std::errc()) {
+    return result;
+  }
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDFilteredView.cc:228]: Is this incredibly obscure C++17 code really the recommended way to do `atoi()`?

Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc:228]: yep, it is, because `from_chars` is a low-level and offers the best possible performance.

**Comment by vargasa** [comment]
Thank you @ianna for looking into this. Once merged I'll check if it's all we need for `MaterialAnalysis`. Thanks again

**Comment by cvuosalo** [comment]
To answer my question about comparison differences, I see these 28 differences in other PRs, so they are probably not related to this one.

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**